src/integrationTests/autoRefresh.test.ts

```diff
@@ -290,6 +290,60 @@ describe("when auto refreshing a non-expired identity which requires a refresh",
       (expect(uid2) as any).toBeInUnavailableState();
     });
   });
+
+  describe("when a new token is set using setIdentity", () => {
+    const manualSetIdentity = makeIdentity({
+      advertising_token: "manual_set_advertising_token",
+    });
+    beforeEach(() => {
+      uid2.setIdentity(manualSetIdentity);
+      getAdvertisingTokenPromise = uid2.getAdvertisingTokenAsync();
+    });
+
+    test("should abort the refreshing request", () => {
+      expect(xhrMock.abort).toHaveBeenCalledTimes(1);
+    });
+
+    test("should reject the getAdvertisingTokenPromise", async () => {
```

Comment:
Why should it reject the promise? If something is waiting for the identity, and it's just been set, I would think the promise should be resolved with the new identity.